# Introduction
![Visitor count](https://shields-io-visitor-counter.herokuapp.com/badge?page=ostash-group.rRNADif&style=for-the-badge)
![GitHub](https://img.shields.io/github/license/ostash-group/rRNADif?style=for-the-badge)
![GitHub release (latest by date including pre-releases)](https://img.shields.io/github/v/release/ostash-group/rRNADif?include_prereleases&style=for-the-badge)
![GitHub issues](https://img.shields.io/github/issues/ostash-group/rRNADif?style=for-the-badge)
![GitHub Repo stars](https://img.shields.io/github/stars/ostash-group/rRNADif?style=for-the-badge)
![GitHub contributors](https://img.shields.io/github/contributors/ostash-group/rRNADif?style=for-the-badge)

![DOI](http://img.shields.io/badge/DOI-10.5281%20%2F%20zenodo.4390532-blue.svg?style=for-the-badge)

rRNADif is a method to measure intragenomic 16S rRNA variability within the desired genome sequence against a chosen set of organisms. The app is calculating ML phylogenetic tree and uses mean/median branch length within set of 16S rRNA as a measure of variability between them <br>
By default precomputed database of 21000+ fully sequenced bacterial genomes is available for comparison. However, custom database generation is possible (see [documentation](https://ostash-group.github.io/rRNADif/))<br>
If you want to contribute, please feel free to do so !(: <br>
**Documentation and user guides are available in the [website](https://ostash-group.github.io/rRNADif/)**
# Analysis Pipeline
![pipeline](images/pipeline.png)
# Citation
This is work in progress, so no publication is available at the moment. If you found this approach valuable, you can site this repo: <br>
P. Hrab & B. Ostash 2020: rRNADif, GitHub repository: https://github.com/pavlohrab/rRNADif

The latest doi is [![DOI](https://zenodo.org/badge/314172997.svg)](https://zenodo.org/badge/latestdoi/314172997)
