<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.68">
<title data-react-helmet="true">Create custom database and run analysis with it | rRNADif</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_language" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Create custom database and run analysis with it | rRNADif"><meta data-react-helmet="true" name="description" content="Create a database"><meta data-react-helmet="true" property="og:description" content="Create a database"><meta data-react-helmet="true" property="og:url" content="https://ostash-group.github.io/rRNADif/docs/database_creation"><link data-react-helmet="true" rel="shortcut icon" href="/rRNADif/img/logo.gif"><link data-react-helmet="true" rel="canonical" href="https://ostash-group.github.io/rRNADif/docs/database_creation"><link rel="stylesheet" href="/rRNADif/styles.d92ff6c2.css">
<link rel="preload" href="/rRNADif/styles.10f63f0c.js" as="script">
<link rel="preload" href="/rRNADif/runtime~main.e14995bc.js" as="script">
<link rel="preload" href="/rRNADif/main.af0ee451.js" as="script">
<link rel="preload" href="/rRNADif/1.2c8a6883.js" as="script">
<link rel="preload" href="/rRNADif/36.bafad98a.js" as="script">
<link rel="preload" href="/rRNADif/37.856a420a.js" as="script">
<link rel="preload" href="/rRNADif/935f2afb.8bec5a8c.js" as="script">
<link rel="preload" href="/rRNADif/35.64ca75e1.js" as="script">
<link rel="preload" href="/rRNADif/88979cd5.f7c2b5f2.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_2AhQ">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/rRNADif/"><img src="/rRNADif/img/logo.gif" alt="My Site Logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img alt="My Site Logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">rRNADif</strong></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/rRNADif/docs/">Docs</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a><div class="react-toggle react-toggle--disabled displayOnlyInLargeViewport_2aTZ"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_BsTx">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_BsTx">ðŸŒž</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" disabled="" aria-label="Dark mode toggle" class="react-toggle-screenreader-only"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/rRNADif/"><img src="/rRNADif/img/logo.gif" alt="My Site Logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img alt="My Site Logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">rRNADif</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/rRNADif/docs/">Docs</a></li><li class="menu__list-item"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_2gpo"><div class="docSidebarContainer_3_JD" role="complementary"><div class="sidebar_2urC"><div class="menu menu--responsive thin-scrollbar menu_5FrY"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_Dm3K" xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 32 32" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Documentation</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/rRNADif/docs/">Introduction</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/rRNADif/docs/installation">Installation and Prerequisites</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Guides</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/get_started">Get started</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/output">Understand the output</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/output2">Create plots and phylogenetic tree</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/add">Non-standard inputs</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/rRNADif/docs/database_creation">Create custom database and run analysis with it</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3EyW"><div class="container padding-vert--lg docItemWrapper_39qw"><div class="row"><div class="col docItemCol_2ASc"><div class="docItemContainer_3QWW"><article><header><h1 class="docTitle_1Lrw">Create custom database and run analysis with it</h1></header><div class="markdown"><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="create-a-database"></a>Create a database<a aria-hidden="true" class="hash-link" href="#create-a-database" title="Direct link to heading">#</a></h2><p>Not only you can take advantage of already created database, but create your own ones! You can use the following steps:</p><ol><li>Download the .csv file from &quot;Browse by genome&quot; tab of NCBI Genome database. (for more information you can refer to the <a href="#get_started.md">get started</a> guide) Alternatively you can use local folder with genome sequences in fasta format.<div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>The .csv file name will be used as a database name. Similarly the folder name will be used for the same purposes if local genome sequences are used</p></div></div></li><li>Run the rRNADif script with <code>--makedb_file</code> (when using .csv file) or <code>--makedb_genomes</code> (for local genomes&#x27; folder) options</li></ol><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif.sh --makedb_file </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">filename.csv</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</h5></div><div class="admonition-content"><p>You can take advantage of <code>--split_download</code> flag while using <code>--makedb_file</code>. In that way, genomes will be downloaded in groups. For each group, genomes will be annotated with barrnap and deleted. Then the next group download begins. Comes in handy when not much space is available for database creation.</p></div></div><ol start="3"><li>(optional) If you want to use a different algorithm for Multiple Sequence Alignment or program for Phylogeny computation feel free to use <code>-m</code> or <code>-t</code> flags. Default programs are mafft for MSA and fasttree for phylogenetic tree computation (GTR model). More options can be viewed via <code>sh rrnadif.sg -h</code> and should be installed before usage (see <a href="/rRNADif/docs/installation">here</a>).</li></ol><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="directory-structure-where-database-is"></a>Directory structure where database is<a aria-hidden="true" class="hash-link" href="#directory-structure-where-database-is" title="Direct link to heading">#</a></h2><p>Databases are stored under datasets folder in the repo. Here, datasets/downloaded_csv folder holds csv file with organism names which were used for database creation and datasets/computed stores the actual data:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database_name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">/</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database_name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">_all.csv </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Dataframe with all pre-computed data. Each row contains genome name, mean, median branch length values</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database_name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">_mean_outliers.csv </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Dataframe that have mean outliers according to the 1,5 IQR</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database_name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">_median_outliers.csv </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Dataframe that have mean outliers according to the 1,5 IQR</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database_name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">_no_ouliers.csv </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Dataframe, with have data with no outliers</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database_name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">_one.csv </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Sequences, which do have only one 16S rRNA (so here is just their names)</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- no_rrna</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">   </span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">files_with_no_rRNAs</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">.fasta </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Files, which had no rRNA annotated at all. Filenames are GenBank assemly names.</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">    </span><span class="token punctuation" style="color:rgb(199, 146, 234)">..</span><span class="token plain">.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- one_rrna</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">   </span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genomes_with_one_16S</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">.fasta </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Genomes which have one 16 rRNA annotated. One fasta file per genome. </span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">    </span><span class="token punctuation" style="color:rgb(199, 146, 234)">..</span><span class="token plain">.</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- rrna</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">   </span><span class="token operator" style="color:rgb(137, 221, 255)">|</span><span class="token plain">-- </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genomes_with_more_than_one_16S</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">.fasta </span><span class="token comment" style="color:rgb(105, 112, 152);font-style:italic"># Genomes which have multiple 16S rRNA annotated. One file per genome.</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">    </span><span class="token punctuation" style="color:rgb(199, 146, 234)">..</span><span class="token plain">.</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>Databases are just csv files. The folders holds fasta files (one genome per file), which will be copied during the analysis if the final phylogenetic tree should be constructed </p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="run-analysis-with-created-database"></a>Run analysis with created database<a aria-hidden="true" class="hash-link" href="#run-analysis-with-created-database" title="Direct link to heading">#</a></h2><p>After the database was created, you should only provide <code>-n</code> flag to specify it (<strong>Database name is a folder name in datasets/computed/</strong>)
Example:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genome.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">  -n </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="update-database"></a>Update database<a aria-hidden="true" class="hash-link" href="#update-database" title="Direct link to heading">#</a></h2><p>The database can be updated using local folder with genomes, or the &quot;Browse by genome&quot; .csv file. Just use proper updatedb flag (<code>--updatedb_file</code> for csv files and <code>--updatedb_genomes</code> for local folder). To specify which database to update please use the <code>-n</code> flag. By default, if no database name specified, the default database will be updated.
Example:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif --updatedb_file </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -n </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">database-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>After the database was created, there is no difference between used sources (csv or local folder). Therefore, multiple sources for updating the same database can be used.</p></div></div><p>Updating the database follows the same logic as the database creation, so you use the same flags to tweak this process. Particularly <code>-m</code> and <code>-t</code> to choose the algorithms and <code>--split_download</code> to specify the number of genomes downloaded at once in case of using csv file. </p><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>Beware of the names of fasta files! By default, all spaces and special characters are replaced by <code>_</code> when the csv file is used. Therefore, to omit duplicate items in the database please use the same convection. </p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="used-rules-to-rename-genomes-with-special-characters-in-their-names"></a>Used rules to rename genomes with special characters in their names<a aria-hidden="true" class="hash-link" href="#used-rules-to-rename-genomes-with-special-characters-in-their-names" title="Direct link to heading">#</a></h2><table><thead><tr><th>Character</th><th>Replaced with</th></tr></thead><tbody><tr><td>Space ( )</td><td>Underscore (_)</td></tr><tr><td>Forward slash (/)</td><td>Underscore (_)</td></tr><tr><td>Colon (:)</td><td>Underscore (_)</td></tr><tr><td>Semicolon (;)</td><td>Underscore (_)</td></tr><tr><td>Comma (,)</td><td>Underscore (_)</td></tr><tr><td>Square brackets ([) or (])</td><td>Deleted ()</td></tr></tbody></table><p>These rules are used when the database is created or updated with the csv file (see [Create a database](#Create a database)). We encourage to follow them when using local files.</p></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/pavlohrab/rRNADif-docs/docs/database_creation.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 40 40" style="margin-right:0.3em;vertical-align:sub"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/rRNADif/docs/add"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Non-standard inputs</div></a></div><div class="pagination-nav__item pagination-nav__item--next"></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_3SO_ thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#create-a-database" class="table-of-contents__link">Create a database</a></li><li><a href="#directory-structure-where-database-is" class="table-of-contents__link">Directory structure where database is</a></li><li><a href="#run-analysis-with-created-database" class="table-of-contents__link">Run analysis with created database</a></li><li><a href="#update-database" class="table-of-contents__link">Update database</a></li><li><a href="#used-rules-to-rename-genomes-with-special-characters-in-their-names" class="table-of-contents__link">Used rules to rename genomes with special characters in their names</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/">Introduction</a></li><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/installation">Installation</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Guides</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/">Get started</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">More</h4><ul class="footer__items"><li class="footer__item"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="text--center"><div>Copyright Â© Pavlo Hrab. Built with Docusaurus.</div></div></div></footer></div>
<script src="/rRNADif/styles.10f63f0c.js"></script>
<script src="/rRNADif/runtime~main.e14995bc.js"></script>
<script src="/rRNADif/main.af0ee451.js"></script>
<script src="/rRNADif/1.2c8a6883.js"></script>
<script src="/rRNADif/36.bafad98a.js"></script>
<script src="/rRNADif/37.856a420a.js"></script>
<script src="/rRNADif/935f2afb.8bec5a8c.js"></script>
<script src="/rRNADif/35.64ca75e1.js"></script>
<script src="/rRNADif/88979cd5.f7c2b5f2.js"></script>
</body>
</html>