<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.68">
<title data-react-helmet="true">Create plots and phylogenetic tree | rRNADif</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_language" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Create plots and phylogenetic tree | rRNADif"><meta data-react-helmet="true" name="description" content="Compute density plot"><meta data-react-helmet="true" property="og:description" content="Compute density plot"><meta data-react-helmet="true" property="og:url" content="https://ostash-group.github.io/rRNADif/docs/output2"><link data-react-helmet="true" rel="shortcut icon" href="/rRNADif/img/logo.gif"><link data-react-helmet="true" rel="canonical" href="https://ostash-group.github.io/rRNADif/docs/output2"><link rel="stylesheet" href="/rRNADif/styles.d92ff6c2.css">
<link rel="preload" href="/rRNADif/styles.10f63f0c.js" as="script">
<link rel="preload" href="/rRNADif/runtime~main.e14995bc.js" as="script">
<link rel="preload" href="/rRNADif/main.af0ee451.js" as="script">
<link rel="preload" href="/rRNADif/1.2c8a6883.js" as="script">
<link rel="preload" href="/rRNADif/36.bafad98a.js" as="script">
<link rel="preload" href="/rRNADif/37.856a420a.js" as="script">
<link rel="preload" href="/rRNADif/935f2afb.8bec5a8c.js" as="script">
<link rel="preload" href="/rRNADif/35.64ca75e1.js" as="script">
<link rel="preload" href="/rRNADif/11406d8f.a56c4f46.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_2AhQ">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/rRNADif/"><img src="/rRNADif/img/logo.gif" alt="My Site Logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img alt="My Site Logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">rRNADif</strong></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/rRNADif/docs/">Docs</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a><div class="react-toggle react-toggle--disabled displayOnlyInLargeViewport_2aTZ"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_BsTx">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_BsTx">ðŸŒž</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" disabled="" aria-label="Dark mode toggle" class="react-toggle-screenreader-only"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/rRNADif/"><img src="/rRNADif/img/logo.gif" alt="My Site Logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img alt="My Site Logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">rRNADif</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/rRNADif/docs/">Docs</a></li><li class="menu__list-item"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_2gpo"><div class="docSidebarContainer_3_JD" role="complementary"><div class="sidebar_2urC"><div class="menu menu--responsive thin-scrollbar menu_5FrY"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_Dm3K" xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 32 32" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Documentation</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/rRNADif/docs/">Introduction</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/rRNADif/docs/installation">Installation and Prerequisites</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Guides</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/get_started">Get started</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/output">Understand the output</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/rRNADif/docs/output2">Create plots and phylogenetic tree</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/add">Non-standard inputs</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/database_creation">Create custom database and run analysis with it</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3EyW"><div class="container padding-vert--lg docItemWrapper_39qw"><div class="row"><div class="col docItemCol_2ASc"><div class="docItemContainer_3QWW"><article><header><h1 class="docTitle_1Lrw">Create plots and phylogenetic tree</h1></header><div class="markdown"><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="compute-density-plot"></a>Compute density plot<a aria-hidden="true" class="hash-link" href="#compute-density-plot" title="Direct link to heading">#</a></h2><p>One of analysis options is a density plot of analyzed database subset values. <strong>The plot is drawn for non-outliers only.</strong>  Example:
<img alt="plot" src="/rRNADif/assets/images/plot-94a80a7cf194856902842b57a19d6b1f.png"></p><p>Just use the <code>--plot</code> flag with no options. (Boolean one)</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genome.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --plot</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>You should install r packages in the environment you are running rRNADif from to draw the plot. Please refer to the <a href="/rRNADif/docs/installation">installation</a> guide for more details.</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="compute-final-phylogenetic-tree"></a>Compute final phylogenetic tree<a aria-hidden="true" class="hash-link" href="#compute-final-phylogenetic-tree" title="Direct link to heading">#</a></h2><p>The main advantage of rRNADif is fast and flexible computation of 16S rRNA based species phylogenetic trees from the comprehensive database. Therefore, no misclassified and/or partly sequenced 16S are included, leading to better species/genus-level classification of genomes based on their rRNAs. As was stated in <a href="/rRNADif/docs/get_started">get started guide</a>, final phylogenetic tree can be computed with <code>--tree</code> flag.</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genome.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --tree</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>The defaults are mafft and fasttree. Also, all sequences are used for outliers and 1 sequence (genome-representative) for non-outlier for giver subset of the database.</p></div></div><p>However, the algorithm for Multiple Sequence alignment or program for phylogeny can be changed with <code>-m</code> and <code>-t</code> flags respectively. For now, rRNADif supports clustalo, muscle and mafft for MSA, and fasttree, iqtree and raxml for phylogeny. </p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genome.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --tree -m muscle -t iqtree</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>By default, branch length for annotated 16S in the input genome will also be computed using provided options, not default ones. Hence, the provided pre-computed database is generated with mafft and fasttree, we suggest to use <code>--only_phylo</code> flag. Therefore, only the final tree will be computed with desired algorithms.   </p></div></div><p>The computed phylogenetic tree will contain every strain and therefore can be simply unreadable. To use only species information and compute species-level phylogenetic tree use <code>--only_species</code> flag.</p><p>In the example below, default options will be used to measure intragenomic variability (construct individual phylogenetic tree) (<code>--only_phylo</code> flag), the muscle for MSA and iqtree for phylogeny (<code>-m</code> and <code>-t</code> flags) will be used for final phylogenetic tree computation. The data for the final phylogenetic tree will contain only species (<code>--only species</code>). This would drastically reduce tree size and respective computation time for it, also making tree human readable. </p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">genome.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --tree -m muscle -t iqtree --only_species --only_phylo</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>Species, when using <code>--only_species</code> flag, will be chosen randomly from the database subset (if multiple strains are available). By default, the first in alphabetical order is kept. Prioritized are genomes with multiple rrnas which are non-outliers.</p></div></div><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>warning</h5></div><div class="admonition-content"><p>If <code>--only_species</code> flag is used, intragenomic variability of input 16S will be compared to the whole subset. Strains, that are the same species will be dropped only for the final tree computation.</p></div></div><p>The output of a run is organized under <code>&lt;project-name&gt;_results/results/phylogeny</code> directory. Three files should be there:</p><ol><li>FINAL.fasta - sequences used for constructing of a phylogenetic tree</li><li>FINAL.mafft - MSA in fasta format. The extension .mafft is used for the sake of consistency in outputs from different MSA programs. This is <strong>NOT</strong> an indicator of the used algorithm.</li><li>FINAL.nwk - phylogenetic tree in newick format. Ready to view in almost every phylogenetic tree viewer.</li></ol></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/pavlohrab/rRNADif-docs/docs/output2.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 40 40" style="margin-right:0.3em;vertical-align:sub"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/rRNADif/docs/output"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Understand the output</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/rRNADif/docs/add"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Non-standard inputs Â»</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_3SO_ thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#compute-density-plot" class="table-of-contents__link">Compute density plot</a></li><li><a href="#compute-final-phylogenetic-tree" class="table-of-contents__link">Compute final phylogenetic tree</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/">Introduction</a></li><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/installation">Installation</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Guides</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/">Get started</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">More</h4><ul class="footer__items"><li class="footer__item"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="text--center"><div>Copyright Â© Pavlo Hrab. Built with Docusaurus.</div></div></div></footer></div>
<script src="/rRNADif/styles.10f63f0c.js"></script>
<script src="/rRNADif/runtime~main.e14995bc.js"></script>
<script src="/rRNADif/main.af0ee451.js"></script>
<script src="/rRNADif/1.2c8a6883.js"></script>
<script src="/rRNADif/36.bafad98a.js"></script>
<script src="/rRNADif/37.856a420a.js"></script>
<script src="/rRNADif/935f2afb.8bec5a8c.js"></script>
<script src="/rRNADif/35.64ca75e1.js"></script>
<script src="/rRNADif/11406d8f.a56c4f46.js"></script>
</body>
</html>