<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.68">
<title data-react-helmet="true">Non-standard inputs | rRNADif</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docusaurus_language" content="en"><meta data-react-helmet="true" name="docusaurus_version" content="current"><meta data-react-helmet="true" name="docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Non-standard inputs | rRNADif"><meta data-react-helmet="true" name="description" content="The rRNADif pipeline consists of 16S annotation, MSA generation, phylogeny computation steps. Each step can be omitted if the proper input is supplied."><meta data-react-helmet="true" property="og:description" content="The rRNADif pipeline consists of 16S annotation, MSA generation, phylogeny computation steps. Each step can be omitted if the proper input is supplied."><meta data-react-helmet="true" property="og:url" content="https://ostash-group.github.io/rRNADif/docs/add"><link data-react-helmet="true" rel="shortcut icon" href="/rRNADif/img/logo.gif"><link data-react-helmet="true" rel="canonical" href="https://ostash-group.github.io/rRNADif/docs/add"><link rel="stylesheet" href="/rRNADif/styles.d92ff6c2.css">
<link rel="preload" href="/rRNADif/styles.10f63f0c.js" as="script">
<link rel="preload" href="/rRNADif/runtime~main.e14995bc.js" as="script">
<link rel="preload" href="/rRNADif/main.af0ee451.js" as="script">
<link rel="preload" href="/rRNADif/1.2c8a6883.js" as="script">
<link rel="preload" href="/rRNADif/36.bafad98a.js" as="script">
<link rel="preload" href="/rRNADif/37.856a420a.js" as="script">
<link rel="preload" href="/rRNADif/935f2afb.8bec5a8c.js" as="script">
<link rel="preload" href="/rRNADif/35.64ca75e1.js" as="script">
<link rel="preload" href="/rRNADif/028affd9.2272dd42.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav aria-label="Skip navigation links"><button type="button" tabindex="0" class="skipToContent_2AhQ">Skip to main content</button></nav><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/rRNADif/"><img src="/rRNADif/img/logo.gif" alt="My Site Logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img alt="My Site Logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">rRNADif</strong></a><a aria-current="page" class="navbar__item navbar__link navbar__link--active" href="/rRNADif/docs/">Docs</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub</a><div class="react-toggle react-toggle--disabled displayOnlyInLargeViewport_2aTZ"><div class="react-toggle-track"><div class="react-toggle-track-check"><span class="toggle_BsTx">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_BsTx">ðŸŒž</span></div></div><div class="react-toggle-thumb"></div><input type="checkbox" disabled="" aria-label="Dark mode toggle" class="react-toggle-screenreader-only"></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/rRNADif/"><img src="/rRNADif/img/logo.gif" alt="My Site Logo" class="themedImage_2E_h themedImage--light_AouX navbar__logo"><img alt="My Site Logo" class="themedImage_2E_h themedImage--dark_1YPN navbar__logo"><strong class="navbar__title">rRNADif</strong></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link navbar__link--active" href="/rRNADif/docs/">Docs</a></li><li class="menu__list-item"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="menu__link">GitHub</a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_2gpo"><div class="docSidebarContainer_3_JD" role="complementary"><div class="sidebar_2urC"><div class="menu menu--responsive thin-scrollbar menu_5FrY"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_Dm3K" xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 32 32" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Documentation</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/rRNADif/docs/">Introduction</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/rRNADif/docs/installation">Installation and Prerequisites</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">Guides</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/get_started">Get started</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/output">Understand the output</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/output2">Create plots and phylogenetic tree</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/rRNADif/docs/add">Non-standard inputs</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/rRNADif/docs/database_creation">Create custom database and run analysis with it</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3EyW"><div class="container padding-vert--lg docItemWrapper_39qw"><div class="row"><div class="col docItemCol_2ASc"><div class="docItemContainer_3QWW"><article><header><h1 class="docTitle_1Lrw">Non-standard inputs</h1></header><div class="markdown"><p>The rRNADif pipeline consists of 16S annotation, MSA generation, phylogeny computation steps. Each step can be omitted if the proper input is supplied. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="use-set-of-16s-rrnas"></a>Use set of 16S rRNAs<a aria-hidden="true" class="hash-link" href="#use-set-of-16s-rrnas" title="Direct link to heading">#</a></h2><p>The set of 16S sequences in fasta format can be used as an input (substituting the genome sequence). In that case, the barrnap annotation step will be omitted. The only flag that is meant to be passed is <code>--step 2</code>.
Example</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-set.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --step </span><span class="token number" style="color:rgb(247, 140, 108)">2</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>warning</h5></div><div class="admonition-content"><p>Please do not pass sequences with identical names - the program will crash. Please also use the following names of sequences:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name_1</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name_2</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><p>So <strong>one of the sequence names must be shorter</strong> than the rest of them (If they are the length program will crash). This is known bug -&gt; we are working to resolve it as soon as possible!</p></div></div><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</h5></div><div class="admonition-content"><p>You can also provide the fasta file with a set of not-related 16S sequences from different genomes. In this case, the rRNADif results about intragenomic variability would have no sense, but the program itself can be used as a quick pipeline to build <strong>high quality 16S phylogenetic trees</strong>. More information is in the tree building <a href="#output2.md">guide</a></p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="use-multiple-sequence-alignment-results"></a>Use Multiple sequence alignment results<a aria-hidden="true" class="hash-link" href="#use-multiple-sequence-alignment-results" title="Direct link to heading">#</a></h2><p>The results of the multiple sequence alignment should be in fasta format. The <code>--step 3</code> flag should be also supplied.</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-MSA.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --step </span><span class="token number" style="color:rgb(247, 140, 108)">3</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><p>If the point of the analysis is to build the final phylogenetic tree, then the sequences of 16S rRNA should be also passed via <code>-s</code> flag</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-MSA.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --step </span><span class="token number" style="color:rgb(247, 140, 108)">3</span><span class="token plain"> -s </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-set.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>warning</h5></div><div class="admonition-content"><p>Please do not pass sequences with identical names - the program will crash. Please also use the following names of sequences:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name_1</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name_2</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><p>So <strong>one of the sequence names must be shorter</strong> than the rest of them (If they are the length program will crash). This is known bug -&gt; we are working to resolve it as soon as possible!
<strong>This warning is for both: MSA file and 16S sequences file</strong></p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="use-phylogenetic-tree-in-newick-format"></a>Use phylogenetic tree in newick format<a aria-hidden="true" class="hash-link" href="#use-phylogenetic-tree-in-newick-format" title="Direct link to heading">#</a></h2><p>You can provide the computed phylogenetic tree as an input in newick format (.nwk files): </p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-tree.nwk</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --step </span><span class="token number" style="color:rgb(247, 140, 108)">4</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><p>If you want to compute the final phylogenetic tree please provide sequences in fasta format via <code>-s</code> flag:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token function" style="color:rgb(130, 170, 255)">sh</span><span class="token plain"> rrnadif -d </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">csv.file</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -i </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-tree.nwk</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> -p </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">project-name</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain"> --step </span><span class="token number" style="color:rgb(247, 140, 108)">4</span><span class="token plain"> -s </span><span class="token operator" style="color:rgb(137, 221, 255)">&lt;</span><span class="token plain">16S-set.fasta</span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><div class="admonition admonition-warning alert alert--danger"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M5.05.31c.81 2.17.41 3.38-.52 4.31C3.55 5.67 1.98 6.45.9 7.98c-1.45 2.05-1.7 6.53 3.53 7.7-2.2-1.16-2.67-4.52-.3-6.61-.61 2.03.53 3.33 1.94 2.86 1.39-.47 2.3.53 2.27 1.67-.02.78-.31 1.44-1.13 1.81 3.42-.59 4.78-3.42 4.78-5.56 0-2.84-2.53-3.22-1.25-5.61-1.52.13-2.03 1.13-1.89 2.75.09 1.08-1.02 1.8-1.86 1.33-.67-.41-.66-1.19-.06-1.78C8.18 5.31 8.68 2.45 5.05.32L5.03.3l.02.01z"></path></svg></span>warning</h5></div><div class="admonition-content"><p>Please do not pass sequences with identical names - the program will crash. Please also use the following names of sequences: </p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><div tabindex="0" class="prism-code language-bash codeBlock_3iAC thin-scrollbar"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name_1</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain"></span><span class="token operator" style="color:rgb(137, 221, 255)">&gt;</span><span class="token plain">Species-name_2</span></div></div></div><button type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button></div></div><p>So <strong>one of the sequence names must be shorter</strong> than the rest of them (If they are the length program will crash). This is known bug -&gt; we are working to resolve it as soon as possible!
<strong>This warning is for both: .nwk file and 16S sequences file</strong></p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="choose-a-database-to-analyse-against"></a>Choose a database to analyse against<a aria-hidden="true" class="hash-link" href="#choose-a-database-to-analyse-against" title="Direct link to heading">#</a></h2><p>By default, rRNADif will use the pre-computed database of 21000+ complete bacterial genomes (Bacteria_full). But custom databases could be <a href="#database_creation.md">constructed</a> and therefore used in the analysis. To change the database <code>-n</code> flag can be used, followed by database name.</p><div class="admonition admonition-note alert alert--secondary"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"></path></svg></span>note</h5></div><div class="admonition-content"><p>For more see database creation <a href="#database_creation.md">guide</a>).</p></div></div></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/pavlohrab/rRNADif-docs/docs/additional_inputs.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 40 40" style="margin-right:0.3em;vertical-align:sub"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/rRNADif/docs/output2"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Create plots and phylogenetic tree</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/rRNADif/docs/database_creation"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Create custom database and run analysis with it Â»</div></a></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_3SO_ thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#use-set-of-16s-rrnas" class="table-of-contents__link">Use set of 16S rRNAs</a></li><li><a href="#use-multiple-sequence-alignment-results" class="table-of-contents__link">Use Multiple sequence alignment results</a></li><li><a href="#use-phylogenetic-tree-in-newick-format" class="table-of-contents__link">Use phylogenetic tree in newick format</a></li><li><a href="#choose-a-database-to-analyse-against" class="table-of-contents__link">Choose a database to analyse against</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><h4 class="footer__title">Docs</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/">Introduction</a></li><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/installation">Installation</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">Guides</h4><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/rRNADif/docs/">Get started</a></li></ul></div><div class="col footer__col"><h4 class="footer__title">More</h4><ul class="footer__items"><li class="footer__item"><a href="https://github.com/ostash-group/rRNADif" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub</a></li></ul></div></div><div class="text--center"><div>Copyright Â© Pavlo Hrab. Built with Docusaurus.</div></div></div></footer></div>
<script src="/rRNADif/styles.10f63f0c.js"></script>
<script src="/rRNADif/runtime~main.e14995bc.js"></script>
<script src="/rRNADif/main.af0ee451.js"></script>
<script src="/rRNADif/1.2c8a6883.js"></script>
<script src="/rRNADif/36.bafad98a.js"></script>
<script src="/rRNADif/37.856a420a.js"></script>
<script src="/rRNADif/935f2afb.8bec5a8c.js"></script>
<script src="/rRNADif/35.64ca75e1.js"></script>
<script src="/rRNADif/028affd9.2272dd42.js"></script>
</body>
</html>